### ISSUE TITLE
Incorrect definite integration of simple exponential involving pi

### ISSUE THREAD

**Issue by gunvirranu** [sympy/sympy#20360]
This is on version 1.6.2. It seems a relatively simple integral involving an exponential gives the incorrect answer.

```python
t, tau, q = symbols("t tau q", real=True)
n = symbols("n", integer=True)
lam = pi * (n - S.Half)

# This gives the correct result
integrate(exp(q * tau), (tau, 0, t)).subs(q, lam)

# This however gives just `t`, which is *not* the same as above
integrate(exp(lam * tau), (tau, 0, t))
```

The result of the second integration is `t` if `lam = 0`, but in this case, it's clearly not. I'm not sure _where_ it ends up making the assumption that `lam == 0`. Note, if I take out the `pi` from `lam`, the two give the same result. Now of course, my math may be way off here and I might be wrong, but it doesn't seem like it.

I briefly looked around for similar issues, but wasn't able to find any. Thanks!

**Comment by oscarbenjamin** [comment]
Thanks for reporting this. It looks like a bug in Piecewise, specifically in the `_handle_irel` method. For this particular case it can be fixed with
```diff
diff --git a/sympy/functions/elementary/piecewise.py b/sympy/functions/elementary/piecewise.py
index 7b8eea208e..47eee0b528 100644
--- a/sympy/functions/elementary/piecewise.py
+++ b/sympy/functions/elementary/piecewise.py
@@ -436,7 +436,7 @@ def _handle_irel(self, x, handler):
             # take them in the order obtained
             args = [(e, args[e]) for e in uniq(exprinorder)]
             # add in the last arg
-            args.append((expr, True))
+            args.append((expr, S.true))
             # if any condition reduced to True, it needs to go last
             # and there should only be one of them or else the exprs
             # should agree
@@ -446,10 +446,11 @@ def _handle_irel(self, x, handler):
                 e, c = args[-1]
                 args[-1] = (e, S.true)
             else:
-                assert len({e for e, c in [args[i] for i in trues]}) == 1
-                args.append(args.pop(trues.pop()))
-                while trues:
-                    args.pop(trues.pop())
+                #assert len({e for e, c in [args[i] for i in trues]}) == 1
+                #args.append(args.pop(trues.pop()))
+                #while trues:
+                #    args.pop(trues.pop())
+                pass
             return Piecewise(*args)
 
     def _eval_integral(self, x, _first=True, **kwargs):
```
However I'm not sure I really understand the implications of that change in general.

**Comment by oscarbenjamin** [comment]
Fixed by #20518.
